sdk/ml/azure-ai-ml/azure/ai/ml/_artifacts/_constants.py

```diff
@@ -40,3 +40,4 @@
 BLOB_STORAGE_CLIENT_NAME = "BlobStorageClient"
 GEN2_STORAGE_CLIENT_NAME = "Gen2StorageClient"
 DEFAULT_CONNECTION_TIMEOUT = 14400
+STORAGE_URI_REGEX = r"(https:\/\/([a-zA-Z0-9@:%_\\\-+~#?&=]+)[a-zA-Z0-9@:%._\\\-+~#?&=]+\.?)\/([a-zA-Z0-9@:%._\\\-+~#?&=]+)\/(.*)"
```

Comment:
1. Where'd you get the storage URI regex
2. Against whose-controlled storage are you using this regex? Will this work for custom domains or PL storage or any of the other edge cases?

Reply:
I took the same regex that we are using in the portal, not sure I understood the question, may be giving an example of the storage uri would be helpful

Reply:
Yes, so are we trying to parse any kind of storage URI or a specific kind we know has restrictions?
For example private link looks like: `"StorageAccountA.privatelink.blob.core.windows.net"`

Reply:
If we know these are always public storage, no custom DNS storage URIs then I think we are good~

Reply:
The regular expression can parse StorageAccountA.privatelink.blob.core.windows.net format also. Thanks for the example.